miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleRestController.java

```diff
@@ -210,8 +210,9 @@ public SampleDto createSample(@RequestBody SampleDto sampleDto, UriComponentsBui
    * 
    * @param sampleDto
    * @return
+   * @throws IOException
    */
-  public Sample buildHierarchy(SampleDto sampleDto) {
+  public Sample buildHierarchy(SampleDto sampleDto) throws IOException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)